socket_test.go

```diff
@@ -220,3 +221,37 @@ func TestConnReaperDeadlock(t *testing.T) {
 		clients[i].Close()
 	}
 }
+
+func TestSocketSendSubscriptionOnConnect(t *testing.T) {
+	endpoint := "inproc://test-resub"
+
+	sub := zmq4.NewSub(context.Background())
+	defer sub.Close()
+	pub := zmq4.NewPub(context.Background())
+	defer pub.Close()
+	sub.SetOption(zmq4.OptionSubscribe, "test")
+	if err := sub.Listen(endpoint); err != nil {
+		t.Errorf("Sub Dial failed: %v", err)
```

Comment:
```suggestion
		t.Failf("Sub Dial failed: %v", err)
```
(no point in going through with the rest of the test)

Reply:
I guess you mean `t.Fatalf` ?